sds/src/event_json.rs

```diff
@@ -0,0 +1,182 @@
+        let first_segment = path.segments.first().unwrap();
+        let mut remaining_segments = path.segments.clone();
+        remaining_segments.remove(0);
+        if let PathSegment::Field(field) = first_segment {
+            let value = self.get_mut(&field.to_string()).unwrap();
+            value.visit_string_mut(&Path::from(remaining_segments), &mut visit);
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use std::collections::HashMap;
+
+    use serde_json::{json, Map, Value};
+
+    use crate::VisitStringResult;
+
+    use super::*;
+
+    #[derive(Debug, PartialEq, Eq)]
+    enum VisitOp {
+        Push(PathSegment<'static>),
+        Pop,
+        Visit(String),
+    }
+
+    struct Visitor {
+        path: Path<'static>,
+        ops: Vec<VisitOp>,
+    }
+
+    impl<'path> EventVisitor<'path> for Visitor {
+        fn push_segment(&mut self, segment: PathSegment<'path>) {
+            self.ops.push(VisitOp::Push(segment.into_static()));
+        }
+
+        fn pop_segment(&mut self) {
+            self.ops.push(VisitOp::Pop);
+        }
+
+        fn visit_string<'s>(&'s mut self, value: &str) -> VisitStringResult<'s, 'path> {
+            self.ops.push(VisitOp::Visit(value.to_string()));
+            VisitStringResult {
+                might_mutate: true,
+                path: &self.path,
+            }
+        }
+    }
+
+    #[test]
+    pub fn test_hashmap_event() {
+        let mut map = Map::new();
+        map.insert(
+            "key-a-1".to_string(),
+            Value::String("value-a-1".to_string()),
+        );
+        map.insert(
+            "key-a-2".to_string(),
+            Value::String("value-b-1".to_string()),
+        );
+        map.insert("key-a-3".to_string(), json!(["an", "array"]));
+        let mut event = HashMap::from([("key-a".to_string(), Value::Object(map))]);
+
+        let mut visitor = Visitor {
+            ops: vec![],
+            path: Path::root(),
+        };
+        event.visit_event(&mut visitor);
+
+        assert_eq!(
+            visitor.ops,
+            vec![
+                VisitOp::Push(PathSegment::Field("key-a".into())),
+                VisitOp::Push(PathSegment::Field("key-a-1".into())),
+                VisitOp::Visit("value-a-1".into()),
+                VisitOp::Pop,
+                VisitOp::Push(PathSegment::Field("key-a-2".into())),
+                VisitOp::Visit("value-b-1".into()),
+                VisitOp::Pop,
+                VisitOp::Push(PathSegment::Field("key-a-3".into())),
+                VisitOp::Push(PathSegment::Index(0)),
+                VisitOp::Visit("an".into()),
+                VisitOp::Pop,
+                VisitOp::Push(PathSegment::Index(1)),
+                VisitOp::Visit("array".into()),
+                VisitOp::Pop,
+                VisitOp::Pop,
+                VisitOp::Pop,
+            ]
+        );
+
+        let mut leaf = String::new();
```

Comment:
Can you add more comments here ? I don't understand why we are getting leaf == "array".to_string()

Reply:
[Add comment to better explain a test](https://github.com/DataDog/dd-sensitive-data-scanner/pull/145/commits/8db4161c844dc576fd753b8a99dd8eba3b69dea8)

Reply:
[Add another comment](https://github.com/DataDog/dd-sensitive-data-scanner/pull/145/commits/1dc6846187202db4f24d013fde9d80e97ef56efb)